cocos/2d/CCNode.h

```diff
@@ -1881,9 +1895,17 @@ class CC_DLL Node : public Ref
     mutable bool _additionalTransformDirty; ///< transform dirty ?
     bool _transformUpdated;         ///< Whether or not the Transform object was updated since the last frame
 
-    int _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
+    union {
+        struct {
+            std::int32_t z; // The original localZOrder
+            std::uint32_t a; // Order Of Arrival, for avoid sort problem with unstable_sort algorithm.
+        } detail;
+        std::int64_t value; // The value to be used in sort
+    } _localZOrder;               ///< Local order (relative to its siblings) used to sort the node
```

Comment:
Yes, you are right, That's why I storage as a unsigned long long and simulate a int32 storage as follow:
```
#define CC_LOCALZORDER_ZERO 0x7fffffffULL
class SimpleNode
    {
    public:
        SimpleNode(int z, unsigned int a)
        {
            _localZOrder = 0;
            setLocalZOrder(z);
            setOrderOfArrival(a);
        }

        int getLocalZOrder() const
        {
            return static_cast<int>(((_localZOrder >> 32) & 0xffffffff) - CC_LOCALZORDER_ZERO);
        }

        unsigned int getOrderOfArrival() const
        {
            return static_cast<unsigned int>((_localZOrder) & 0xffffffff);
        }

        void setLocalZOrder(int z)
        {
            _localZOrder = ((CC_LOCALZORDER_ZERO + z) << 32) | (_localZOrder & 0xffffffff);
        }

        void setOrderOfArrival(unsigned int orderOfArrival)
        {
            _localZOrder = (_localZOrder & 0xffffffff00000000) | orderOfArrival;
        }
        
        std::uint64_t _localZOrder;
    };
```